### ISSUE TITLE
Need to change default value of fates_comp_excln

### ISSUE THREAD

**Issue by ckoven** [NGEET/fates#303]
So I've been playing around with some runs where I started from BCI inventory (using datasets described in #297), using the full FATES (i.e. not prescribed physiology mode).  A thing that happens if I do this run with default parameter files is that the initial set of trees all die out in the first few years, and then subsequently get replaced by recruits that grow up and fill in the forest.  What this looks like, if I plot basal area per size bin over time, is the following:

![control_case](https://user-images.githubusercontent.com/10852790/33087007-3b03294c-ce9e-11e7-8a44-7d712a9a4b6c.png)

I did some experiments to try to figure out what was going on.  A clue is that canopy mortality is never high, which means that that initial set of inventory-initted trees is dying because they are in the understory, even though they are large.  But that doesn't make sense, as they ought to stay in the canopy, since they are large.  So I tried playing around with the fates_comp_excln parameter to see if that fixed things. For those who haven't played with this, what it does is control the probability of being in the canopy versus the understory as controlled by DBH (it should be based on height of course, but that's a separate issue, #302).  In code, what this looks like for the case of promotion is: `currentCohort%prom_weight = currentCohort%dbh**ED_val_comp_excln`. First I set the parameter to a value of -1, which triggers the deterministic PPA algorithm instead of the stochastic PPA algorithm for sorting.  That changed the behavior completely, so that it now looked like this:

![determ_ppa_case](https://user-images.githubusercontent.com/10852790/33087025-4376028e-ce9e-11e7-8b36-9aa614156d8a.png)

So from that I concluded that what's going on in the control case is that the number of tiny trees is so large, that they are actually ending up in the canopy relative to the large trees that are already there, which then kills off the big trees.  So obviously this wouldn't make sense that tiny trees are outcompeting the really big trees for light.  But I wanted to see if the stochastic PPA always produced this behavior, so I tried to intermediate cases, with fates_comp_excln set to 1.0 and 3.0.

The fates_comp_excln = 1.0 case looks like this:

![stoch_ppa_cmplexcl_1 0](https://user-images.githubusercontent.com/10852790/33087045-53b1d240-ce9e-11e7-82a9-11ba9684c9e4.png)

And the fates_comp_excln = 3.0 case looks like this:

![stoch_ppa_cmplexcl_3 0](https://user-images.githubusercontent.com/10852790/33087061-5988324a-ce9e-11e7-83e2-a7c3e3c41ba8.png)

So there are a few interesting things going on here.  (1) is that the default value is too low.  Seedlings shouldn't be able to outcompete tall trees for light, no matter how many there are.  I've noticed this behavior in other runs, but the fact that it shows up here means that we really need to increase the default value of that parameter.  (2) is that the deterministic PPA case and the stochastic but high value of fates_comp_excln are qualitatively similar (in this configuration), which is good and makes sense.  (3) is that intermediate values of the parameter like the 1.0 case, smaller trees are still on a relatively even playing field with their larger neighbors, on account of their larger number density.  (4) is that this parameter, even in a single-PFT configuration, has enormous power. However, it is pretty much unconstrained by direct observations, and the way it works when it is differing qualitatively from the deterministic case doesn't really make physical sense, as it implies that all of the tall trees are either stacked one on top of each other or else actually underneath the shorter trees. (5) is that, with either a big value of the parameter (fates_comp_excln=3.0) or the deterministic PPA case (fates_comp_excln=-1), the final size distribution isn't so different from the initial values that we're getting from the inventory, which is somewhat reassuring.

Anyway, I just wanted to raise this as an issue so that people are aware of it, and in particular if they are wondering why all the big trees die after an inventory initialization with the default parameters, this is why.  I'd also recommend that we move towards a default value of the parameter that is larger than 1 for the time being.  I believe that there is a whole other set of issues with this parameter when doing multiple PFT competitive experiments, but thats also a separate issue that we need to explore further to understand.  For now, though, just based on single-PFT dynamics, I feel like we can rule out values of the parameter that are less than 1.

**Comment by glemieux** [comment]
Talking with @JessicaNeedham about #828, we wondered if the bug fix for the `canopy_layer_tlai` might impact this behavior.  :thinking:

**Comment by ckoven** [comment]
Yes, in principle, if we switch this to use the deterministic algorithm, then I think that the problem that #828 fixes would be less evident. But still better to have #828 in place.
